Add compatibility layer for Linux kernel versions

AntoineBlaud opened the pull request:
## Compatibility Layer for Linux Kernel Changes

### Overview

This pull request introduces a compatibility layer to address structural changes in the Linux kernel starting from version 5.17. Notably, modifications have been made to the `page` and `slab` structures, necessitating adjustments in codebase.

AntoineBlaud committed:
Add compatibility layer for Linux kernel versions

fidgetingbits commented:
Thanks! I'll try to find time to review and test it tomorrow.

fidgetingbits commented on libslub/frontend/commands/gdb/sbtrace.py:
```diff
@@ -56,6 +58,10 @@ def invoke(self, arg, from_tty):
 
         log.debug("sbtrace.invoke()")
+        
+        if len(self.args.names) == 0:
```
I guess this is related to a separate fix. I don't mind to keep it but you can feel free to send a separate PR if you want.

fidgetingbits commented on libslub/pyslub.py:
```diff
@@ -35,8 +37,12 @@ def __init__(self):
+            
+        # read the current kernel version using gdb
+        version_string = gdb.parse_and_eval("(char *)linux_banner").string()
+        release = re.search(r'(\d+\.\d+\.\d+)', version_string).group(1)
```
Given that this is duplicated code from earlier probably just makes sense to make it like a helper function or something and call it in both places. I don't mind to do it later if you don't want to do it now though.

AntoineBlaud replied:
Hello @fidgetingbits,

Following your guidance, I've created a second commit to address the duplicate code issue.

fidgetingbits reviewed (approved):
This seems to work great. I've tested it on something earlier and later then 5.17. I'll wait a little bit to see if you respond about wanting to use a single helper function, but if you don't want to change that I will merge it as is and address it later.

AntoineBlaud committed:
Refactored kernel_compatibility_layer.py:

    Renamed the file to kernel_compat_layer.py.
    Moved the function get_kernel_version() outside of the class KernelCompatibilityLayer.
    Updated references across the codebase to reflect the changes.
